locator/abstract_replication_strategy.cc

```diff
@@ -187,6 +187,9 @@ abstract_replication_strategy::do_get_ranges(inet_address ep, const token_metada
             }
         }
         prev_tok = tok;
+        if (can_yield) {
+            seastar::thread::maybe_yield();
+        }
```

Comment:
@bhalevy please review. I think this is safe since token_metadata_ptr should protect the contents from mutation, but I forgot the details.

Reply:
yes, it is safe. `tmptr` references an immutable copy of the token_metadata and since with `can_yield::yes` we're running in a seastar thread, it will not go away.